### PR TITLE
Disable COM specific marshalers on Unix

### PR DESCRIPTION
**Pull request by jkotas**
Match CoreCLR behavior and prevent compiler from crashing

### CODE REVIEW

**Comment by jkotas** [comment]
`[DllImport("abc")] extern static void f(object o);` was crashing the compiler. Kennedy hit this issue when trying to compile for Linux.

File: src/coreclr/tools/Common/TypeSystem/Interop/IL/MarshalHelpers.cs [hunk 1]
```diff
@@ -266,7 +266,10 @@ public static partial class MarshalHelpers
                                 return MarshallerKind.CBool;
 
                             case NativeTypeKind.VariantBool:
-                                return MarshallerKind.VariantBool;
+                                if (type.Context.Target.IsWindows)
```
Comment (MichalStrehovsky) [inline src/coreclr/tools/Common/TypeSystem/Interop/IL/MarshalHelpers.cs]:
```suggestion
                                if (context.Target.IsWindows)
```

File: src/coreclr/tools/Common/TypeSystem/Interop/IL/MarshalHelpers.cs [hunk 2]
```diff
@@ -580,12 +583,18 @@ public static partial class MarshalHelpers
             {
                 if (nativeType == NativeTypeKind.AsAny)
                     return isAnsi ? MarshallerKind.AsAnyA : MarshallerKind.AsAnyW;
-                else if ((isField && nativeType == NativeTypeKind.Default)
-                    || nativeType == NativeTypeKind.Intf
-                    || nativeType == NativeTypeKind.IUnknown)
-                    return MarshallerKind.ComInterface;
                 else
-                    return MarshallerKind.Variant;
+                if (type.Context.Target.IsWindows)
```
Comment (MichalStrehovsky) [inline src/coreclr/tools/Common/TypeSystem/Interop/IL/MarshalHelpers.cs]:
```suggestion
                if (context.Target.IsWindows)
```

File: src/coreclr/tools/Common/TypeSystem/Interop/IL/MarshalHelpers.cs [hunk 3]
```diff
@@ -641,7 +650,10 @@ public static partial class MarshalHelpers
             }
             else if (type.IsInterface)
             {
-                return MarshallerKind.ComInterface;
+                if (type.Context.Target.IsWindows)
```
Comment (MichalStrehovsky) [inline src/coreclr/tools/Common/TypeSystem/Interop/IL/MarshalHelpers.cs]:
```suggestion
                if (context.Target.IsWindows)
```

**Review by MichalStrehovsky — APPROVED** [review]
Thank you!

**Pull request merged**